x-pack/solutions/search/plugins/search_synonyms/server/routes.ts

```diff
@@ -270,4 +273,158 @@ export function defineRoutes({ logger, router }: { logger: Logger; router: IRout
       });
     })
   );
+
+  router.put(
+    {
+      path: APIRoutes.SYNONYM_SET_ID,
+      options: {
+        access: 'internal',
+        tags: ['synonyms:write', 'synonyms:read'],
+      },
+      security: {
+        authz: {
+          requiredPrivileges: ['synonyms:write', 'synonyms:read'],
```

Comment:
Question: This is totally fine, it matches the privileges that `all` would assign, but why have `synonyms:read` required for a PUT route when you have `write` as well?

Reply:
Not required really technically. But logically these are used in the a synonyms management UI where you would need to read and write at the same time in UI.